torch/csrc/api/include/torch/detail/TensorDataContainer.h

```diff
@@ -0,0 +1,303 @@
+#pragma once
+
+#include <ATen/ATen.h>
+
+#include <initializer_list>
+
+namespace torch {
+
+namespace detail {
+
+enum class TensorDataContainerType { Scalar, InitList, Tensor };
+
+struct TensorDataContainer;
+
+inline std::ostream& operator<<(std::ostream& stream, const TensorDataContainer& tensor_data_container);
+
+// We use `TensorDataContainer` to support converting the following data container types
+// into the equivalent Tensor:
+//
+// 1. Arbitrarily nested braced-init-list (e.g. `{{1, 2}, {3, 4}}`).
+// 2. `at::ArrayRef` of supported tensor data types.
+// 3. `std::vector` of supported tensor data types.
+//
+// At any time, a `TensorDataContainer` object represents one of the following:
+//
+// 1. A scalar with value `scalar()` and type `scalar_type()`.
+// 2. A Tensor represented in `std::initializer_list<TensorDataContainer>` form,
+//    with value `init_list()`, Tensor scalar type `scalar_type()`, and Tensor sizes `sizes()`.
+// 3. A Tensor represented in `at::Tensor` form, with value `tensor()`, scalar type `scalar_type()`,
+//    and Tensor sizes `sizes()`.
+//
+// All the infrastructure here is mostly to support converting an arbitrarily nested braced-init-list
+// to the equivalent Tensor successfully. Consider the following example:
+//
+// `torch::tensor({{1}, {2}})`
+//
+// Here is the code path that it goes through:
```

Comment:
It's a little weird to say "here is the codepath that it goes through" because there is only one torch::tensor overload. Maybe say "this is the torch::tensor declaration"?

Reply:
Fixed